gcloud-java-core/src/main/java/com/google/gcloud/FieldSelector.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud;
+
+import com.google.common.base.Function;
+import com.google.common.base.Joiner;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Interface for Google Cloud resource's fields. Implementations of this interface can be used to
+ * select only desired fields from a returned Google Cloud resource.
+ */
+public interface FieldSelector {
+
+  /**
+   * Returns a string selector. This selector is passed to a Google Cloud service (possibly with
+   * other field selectors) to specify which resource fields should be returned by an API call.
+   */
+  String selector();
+
+  /**
+   * A helper class used to build composite selectors given a number of fields. This class is not
+   * supposed to be used directly by users.
+   */
+  class SelectorHelper {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [All methods are static.  Consider using a utility class instead. Alternatively, you could add a private constructor or make the class abstract to silence this warning.](https://www.codacy.com/app/mziccard/gcloud-java/file/2326222243/issues/source?bid=3123103&fileBranchId=3204660#l44)